### PR TITLE
Draft: Initial buildout of lincbrain CLI/Python Client from dandi-cli

### PR DESCRIPTION
**Pull request by aaronkanzer**
This PR/branch is being used to clone/transform the `dandi-cli` into the `lincbrain` CLI tool.

A couple things:

`versioneer` is still problematic for `lincbrain`, therefore, references are commented out.

The publishing process includes: 

1. Bump the value here: https://github.com/lincbrain/linc-cli/pull/2#discussion_r1406778482

2. Run `python setup.py sdist bdist_wheel` locally -- this will populate a new build value in your local `dist/` folder

3. Run `twine upload dist/lincbrain-<new-semantic-version>*`

4. You'll be prompted for a username and password: for username, use `__token__`, for password, use a PyPI token issued to you (ask @aaronkanzer for one if you don't have it)

This workflow will change, but should be sufficient for now

### CODE REVIEW

File: lincbrain/dandiapi.py [hunk 1]
```diff
@@ -417,7 +417,7 @@ def __init__(
         :param str token: User API Key. Note that different instance APIs have
             different keys.
         """
-        check_dandi_version()
```
Comment (aaronkanzer) [inline lincbrain/dandiapi.py:420]: `check_dandi_version()` (and `versioneer` by extension) still need to be set up correctly -- thus references are commented out for now.

File: lincbrain/utils.py [hunk 2]
```diff
@@ -610,10 +610,10 @@ def _get_instance(
             f" please contact that server's administrators: {e}"
         )
     our_version = Version(__version__)
-    if our_version < minversion:
-        raise CliVersionTooOldError(our_version, minversion, bad_versions)
-    if our_version in bad_versions:
-        raise BadCliVersionError(our_version, minversion, bad_versions)
```
Comment (aaronkanzer) [inline lincbrain/utils.py:616]: Similar to https://github.com/lincbrain/linc-cli/pull/2/files#r1406776675 -- `versioneer` needs to be refined

File: setup.py [hunk 3]
```diff
@@ -25,22 +25,22 @@
...
-    version_config = "0.1.0"  # Fallback version
-    cmdclass = {}
+# try:
+#     import versioneer
+#     version_config = versioneer.get_version()
+#     cmdclass = versioneer.get_cmdclass()
+# except ImportError:
+#     print("WARNING: failed to import versioneer, falling back to no version for now")
+#     version_config = "0.4.0"  # Fallback version
+#     cmdclass = {}
```
Comment (aaronkanzer) [inline setup.py:35]: Similar to https://github.com/lincbrain/linc-cli/pull/2/files#r1406776675 -- versioneer needs to be refined

File: setup.py [hunk 4]
```diff
@@ -25,22 +25,22 @@
...
-    version_config = version_config.split('+')[0]
+# if '+' in version_config:
+#     version_config = version_config.split('+')[0]
 
 if __name__ == "__main__":
     setup(
         name="lincbrain",
-        version="0.1.0",
-        cmdclass=cmdclass,
+        version="0.9.0",
```
Comment (aaronkanzer) [inline setup.py:44]: This is the value that is bumped to create a new semantic version in PyPI for now during the build and distribute phases

**Comment by kabilar** [comment]
Hi @aaronkanzer, let's change the term from `dandiset` to `dataset`.  For example, [here](https://github.com/lincbrain/linc-cli/blob/7eb75fef5a74c2608e779228d4fdeadb4a603f0d/lincbrain/dandiset.py#L79):

```https://dandiarchive.org/dandiset/``` -> ```https://lincbrain.org/dataset/```

**Comment by kabilar** [comment]
Should we change `DANDI_API_KEY` to `LINCBRAIN_API_KEY`?

File: lincbrain/cli/command.py
Comment (kabilar) [inline lincbrain/cli/command.py:1]: Please update [line 100](https://github.com/lincbrain/linc-cli/blob/3e3af7a6bfdccd960f2b35e8113a65e0cee76aa2/lincbrain/cli/command.py#L100) to the following:

```python
logdir = platformdirs.user_log_dir("lincbrain-cli", "lincbrain")
```

**Pull request merged**